### ISSUE TITLE
learned Prior

### ISSUE THREAD

**Issue by dambuck** [tensorflow/models#5680]
Hi,
Its actually not and issue...

after reading the paper I had a look at the code and was surprised to see that the prior is learned aswell. I could not find any information about this process in the related paper.

As I am not really proficient with tensorflow code and mostly just fooling around with keras, I was wondering how the prior is learned? 
The KL loss is computed between prior and posterior, while the posterior is simpy the output of the encoder. Could you explain how the learning is done?

Could I implement it in the form of a random initialized layer and compute the KL between this layer an the posterior? Or might it work to connect the encoder to another layer(additional to the the sampling and feed forward to the generator) and use this layers output as the prior in the KL loss?

Using keras 2.2.2
tf 1.10.1 no gpu support
custom code

Thank you

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
